src/shred_collector/shred_verifier.zig

```diff
@@ -15,30 +14,23 @@ const Packet = sig.net.Packet;
 pub fn runShredVerifier(
     exit: *Atomic(bool),
     /// shred receiver --> me
-    unverified_shred_receiver: *Channel(ArrayList(Packet)),
+    unverified_shred_receiver: *Channel(Packet),
     /// me --> shred processor
-    verified_shred_sender: *Channel(ArrayList(Packet)),
+    verified_shred_sender: *Channel(Packet),
     leader_schedule: SlotLeaderProvider,
 ) !void {
     var verified_count: usize = 0;
-    var buf = ArrayList(ArrayList(Packet)).init(unverified_shred_receiver.allocator);
-    while (!exit.load(.unordered)) {
-        try unverified_shred_receiver.tryDrainRecycle(&buf);
-        if (buf.items.len == 0) {
-            std.time.sleep(10 * std.time.ns_per_ms);
-            continue;
-        }
-        for (buf.items) |packet_batch| {
+    while (!exit.load(.acquire)) {
+        while (unverified_shred_receiver.receive()) |packet| {
             // TODO parallelize this once it's actually verifying signatures
-            for (packet_batch.items) |*packet| {
-                if (!verifyShred(packet, leader_schedule)) {
-                    packet.flags.set(.discard);
-                } else {
-                    verified_count += 1;
-                }
+            var our_packet = packet;
+            if (!verifyShred(&packet, leader_schedule)) {
+                our_packet.flags.set(.discard);
```

Comment:
Without using packet batches, this line could just be `continue`

Reply:
Gotcha.

Reply:
b47ab63